avsomr/ui/taskbarprogress.cpp

```diff
@@ -0,0 +1,116 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+
+#include "taskbarprogress.h"
+
+#ifdef Q_OS_WIN
+#include <QWinTaskbarProgress>
+#endif
+
+using namespace Ms;
+
+TaskbarProgress::TaskbarProgress(QObject* parent)
+      : QObject(parent)
+      {
+#ifdef Q_OS_WIN
+      _winTaskbarProgress = new QWinTaskbarProgress();
+#endif
+      }
+
+TaskbarProgress::~TaskbarProgress()
+      {
+#ifdef Q_OS_WIN
+      delete _winTaskbarProgress;
+#endif
+      }
+
+//---------------------------------------------------------
+//   isAvalabled
+//---------------------------------------------------------
+
+bool TaskbarProgress::isAvalabled() const
+      {
+#ifdef Q_OS_WIN
+      return true;
+#endif
+      return false;
+      }
+
+//---------------------------------------------------------
+//   show
+//---------------------------------------------------------
+
+void TaskbarProgress::show()
+      {
+#ifdef Q_OS_WIN
+      _winTaskbarProgress->show();
+#endif
+      // noop
```

Comment:
What does this `noop` mean?

Reply:
noop - no operation, nothing is done

Reply:
not to be confused with noob ;-)

Reply:
I wonder whether these `// noop` comments are really necessary... Shouldn't it be obvious if in some cases there's no operation?

Reply:
With this comment, I want to explicitly show two things:

* no implementation for other platforms
* nothing bad will happen, this is not a error, just nothing will be done

Reply:
Then I guess this is clearer:
```
#ifdef Q_OS_WIN
      _winTaskbarProgress->show();
#else
      // noop
#endif
```

Reply:
Yes, I agree, already fixed it